### PR TITLE
move upload step top platform specific builds, cleanup

### PR DESCRIPTION
**Pull request by jakubgs**
This is in preparation to switch to building different platforms separately.

### CODE REVIEW

**Comment by ghost** [comment]
### Pull Request Checklist
- [ ] Have you updated the documentation, if impacted (e.g. [docs.status.im](https://docs.status.im/docs/build_status.html))?

**Review by mandrigin — APPROVED** [review]
lgtm

File: ci/Jenkinsfile.android [hunk 1]
```diff
@@ -90,4 +92,7 @@ pipeline {
       }
     }
   }
+  post {
+    failure { script { cmn.gitHubNotifyPRFail() } }
```
Comment (pedropombeiro) [inline ci/Jenkinsfile.android]: Does this ever get called when we're not building for a PR branch (i.e. `env.CHANGE_ID` doesn't have a value)?

**Review by pedropombeiro — APPROVED** [review]
LGTM, added a question for clarification

File: ci/Jenkinsfile.android (hunk 1 above)
Comment (jakubgs) [reply ci/Jenkinsfile.android]: Good catch, yeah, I just caught that, so I added it as an argument the `buildBranch` method. Check it out.

**Review by rasom — APPROVED** [review]

File: ci/Jenkinsfile.linux [hunk 2]
```diff
@@ -24,6 +24,7 @@ pipeline {
 
   options {
     timestamps()
+    disableConcurrentBuilds()
```
Comment (mandrigin) [inline ci/Jenkinsfile.linux:27]: why is that needed?

File: ci/Jenkinsfile.ios [hunk 3]
```diff
@@ -69,12 +70,13 @@ pipeline {
...
     stage('Archive') {
       steps {
-        script { archiveArtifacts api }
+        archiveArtifacts api
       }
     }
     stage('Upload') {
       steps {
         script {
+          env.PKG_URL = cmn.uploadArtifact(api)
```
Comment (mandrigin) [inline ci/Jenkinsfile.ios]: we should upload to diawi instead

Comment (jakubgs) [reply ci/Jenkinsfile.ios]: I can drop it. Sure.

File: ci/Jenkinsfile.linux (hunk 2 above)
Comment (mandrigin) [reply ci/Jenkinsfile.linux:27]: okay, it is only blocking jobs from the same PR, resolving

Comment (jakubgs) [reply ci/Jenkinsfile.linux:27]: The idea is that it affects each PR separately(since each PR has it's own build generated), which means there are no concurrent builds for the same platforms for the same pr.

The idea is to not let someone hog all the builders by pushing a lot of commits at once to a PR.

**Review by mandrigin — APPROVED** [review]

**Comment by oskarth** [comment]
I see now why we are the most active ERC20 project :D

**Comment by jakubgs** [comment]
>I see now why we are the most active ERC20 project :D

Yeah, trying to polish the new build setup for GitHub, generates a lot of spam.
Later we can optimize this by writing a more complex script to post just one comment and keep updating it.
